CRM/Contribute/BAO/Contribution.php

```diff
@@ -5320,12 +5344,16 @@ public static function recordAlwaysAccountsReceivable(&$trxnParams, $contributio
    * @param array $amountParams
    *
    * @param string $context
+   * @param array $fieldValues
    *
    * @return float
    */
-  public static function calculateFinancialItemAmount($params, $amountParams, $context) {
+  public static function calculateFinancialItemAmount($params, $amountParams, $context, &$fieldValues) {
```

Comment:
this feels like a code smell. adding an extra pass by reference at the end is generally a clue to it being a bad practice. Is there another way? Could this be dealt with in the calling function?